IdentityCore/src/cache/MSIDMacKeychainTokenCache.m

```diff
@@ -0,0 +1,660 @@
+    [self createUnimplementedError:error context:context];
+    return FALSE;
+}
+
+#pragma mark - App Metadata
+
+// Save MSIDAppMetadataCacheItem (clientId/environment/familyId) in the macOS keychain cache.
+- (BOOL)saveAppMetadata:(__unused MSIDAppMetadataCacheItem *)item
+                    key:(__unused MSIDCacheKey *)key
+             serializer:(__unused id<MSIDAppMetadataItemSerializer>)serializer
+                context:(__unused id<MSIDRequestContext>)context
+                  error:(__unused NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return FALSE;
+}
+
+// Read MSIDAppMetadataCacheItem (clientId/environment/familyId) items from the macOS keychain cache.
+- (NSArray<MSIDAppMetadataCacheItem *> *)appMetadataEntriesWithKey:(__unused MSIDCacheKey *)key
+                                                        serializer:(__unused id<MSIDAppMetadataItemSerializer>)serializer
+                                                           context:(__unused id<MSIDRequestContext>)context
+                                                             error:(__unused NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return nil;
+}
+
+// Remove items with the given Metadata key from the macOS keychain cache.
+- (BOOL)removeItemsWithMetadataKey:(__unused MSIDCacheKey *)key
+                           context:(__unused id<MSIDRequestContext>)context
+                             error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return FALSE;
+}
+
+#pragma mark - Wipe Info
+
+// Saves information about the app which most-recently removed a token.
+- (BOOL)saveWipeInfoWithContext:(__unused id<MSIDRequestContext>)context
+                          error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return FALSE;
+}
+
+// Read information about the app which most-recently removed a token.
+- (NSDictionary *)wipeInfo:(__unused id<MSIDRequestContext>)context
+                     error:(NSError **)error
+{
+    [self createUnimplementedError:error context:context];
+    return nil;
+}
+
+#pragma mark - clear
+
+// A test-only method that deletes all items from the cache for the given context.
+- (BOOL)clearWithContext:(id<MSIDRequestContext>)context
+                   error:(NSError **)error
+{
+    MSID_TRACE;
+    MSID_LOG_WARN(context, @"Clearing the whole context. This should only be executed in tests");
+
+    // For now, this just deletes all accounts (an empty key matches all account items):
+    MSIDMacKeychainAccountCacheKey* key = [MSIDMacKeychainAccountCacheKey new];
+    return [self removeItemsWithAccountKey:key context:context error:error];
+}
+
+#pragma mark - Utilities
+
+// Get the basic/default keychain query dictionary for account items.
+// To support searching, the key properties may be partially or fully empty.
+- (NSMutableDictionary *)accountQueryForKey:(MSIDCacheKey *)key
```

Comment:
Should this also support creating query by type or generic?

Reply:
Yes, our usage is a little different so I wasn't doing this. I've updated the code.

Reply:
updated usage of keys/queries